### ISSUE TITLE
Feature request: warning for same style equation

### ISSUE THREAD

**Issue by andras-tim** [koalaman/shellcheck#888]
#### For new checks and feature suggestions
- [ ] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [ ] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/bash
foo=0

if [[ "${foo}" == 0 ]]; then
    echo 1
fi

if [[ "${foo}" = 0 ]]; then
    echo 2
fi

if [[ "${foo}" -eq 0 ]]; then
    echo 3
fi
```

#### Here's what shellcheck currently says:
No issues detected!

#### Here's what I wanted or expected to see:
Warning: Please use same style equation

**Comment by kurahaupo** [comment]
I suspect this request is hopelessly optimistic. There's a lot of groundwork needed to add detection of the divergent usage; it isn't a trivial addition.

And you can't merge the last one with the other two:

    foo=00
    [[ $foo = 0 ]] # false
    [[ $foo == 0 ]] # false
    [[ $foo -eq 0 ]] # true

(And the omission of {} and "" makes no difference there.)

I'd prefer to get a recommendation that it be replaced by:

    (( foo == 0 ))

As for the others, using `=` vs `==` inside `[[  ]]` seems to be developing into a holy war. 

The arguments boil down to:
* `==` is the comparison operator all most C-derived languages, not to be confused with `=` assignment;
* there is historical precedent for `=`_ so don't change it; and
* since they're pattern matches rather than comparisons, having a different operator is a _good_ thing.
